lib/sycamore/sycamore/transforms/standardizer.py

```diff
@@ -0,0 +1,121 @@
+from abc import ABC, abstractmethod
+
+from sycamore.plan_nodes import Node
+
+from sycamore.data import Document
+from sycamore.transforms.map import Map
+from dateutil import parser
+
+
+class Standardizer(ABC):
+
+    @abstractmethod
+    def standardize(self, document: Document) -> Document:
+        pass
+
+
+class LocationStandardizer(Standardizer):
+    """
+    This Class standardizes the format of US state location.
+    """
+
+    def __init__(self):
+        self._state_dict = {
+            "AK": "Alaska",
+            "AL": "Alabama",
+            "AR": "Arkansas",
+            "AZ": "Arizona",
+            "CA": "California",
+            "CO": "Colorado",
+            "CT": "Connecticut",
+            "DC": "District of Columbia",
+            "DE": "Delaware",
+            "FL": "Florida",
+            "GA": "Georgia",
+            "HI": "Hawaii",
+            "IA": "Iowa",
+            "ID": "Idaho",
+            "IL": "Illinois",
+            "IN": "Indiana",
+            "KS": "Kansas",
+            "KY": "Kentucky",
+            "LA": "Louisiana",
+            "MA": "Massachusetts",
+            "MD": "Maryland",
+            "ME": "Maine",
+            "MI": "Michigan",
+            "MN": "Minnesota",
+            "MO": "Missouri",
+            "MS": "Mississippi",
+            "MT": "Montana",
+            "NC": "North Carolina",
+            "ND": "North Dakota",
+            "NE": "Nebraska",
+            "NH": "New Hampshire",
+            "NJ": "New Jersey",
+            "NM": "New Mexico",
+            "NV": "Nevada",
+            "NY": "New York",
+            "OH": "Ohio",
+            "OK": "Oklahoma",
+            "OR": "Oregon",
+            "PA": "Pennsylvania",
+            "RI": "Rhode Island",
+            "SC": "South Carolina",
+            "SD": "South Dakota",
+            "TN": "Tennessee",
+            "TX": "Texas",
+            "UT": "Utah",
+            "VA": "Virginia",
+            "VT": "Vermont",
+            "WA": "Washington",
+            "WI": "Wisconsin",
+            "WV": "West Virginia",
+            "WY": "Wyoming",
+        }
+
+    def _standardize_state(self, state: str) -> str:
+        clean_state = state.lstrip().rstrip().upper()
+        return self._state_dict.get(clean_state, "")
+
+    def standardize(self, doc: Document) -> Document:
+        if "location" not in doc.properties["entity"]:
+            return doc
+        raw_location = doc.properties["entity"]["location"]
+        city, state = raw_location.split(",")
```

Comment:
I don't like this very much. It is going to be very brittle. "WA" can also mean Western Australia for example. And you should rarely assume that cities are always represented as "<City name>, <State abbreviation>". This might work in the very narrow case of the NTSB dataset that we have but it is far from general.

I would recommend either looking to see if there is an existing library that can do this better, or put a lot of caveats on the docstring on this class indicating it only works for locations expressed as "city, state" for US state abbreviations.

Reply:
+1 

I think if we're going with very specific implementations I'd rather have this as a function that a script will directly use `(docset.map(thisSpecificTypeOfStandarization)`

Top-level transforms and utilities need to be more resilient